### PR TITLE
Refactor git package

### PR DESCRIPTION
**Pull request by Filbird**
Initial stab at refactoring the git package

### CODE REVIEW

File: launch.go [hunk 1]
```diff
@@ -182,11 +182,23 @@ func launch(api screwdriver.API, buildID string, rootDir string) error {
...
 		return err
 	}
 
+	err = repo.Checkout(b.SHA)
+	if err != nil {
+		return err
+	}
+
+	if pr != "" {
+		err = repo.MergePR(pr, "pr")
```
Comment (jer) [inline launch.go]: `"pr"` here seems unnecessary. Can we just move it into MergePR and make it a single parameter?

Comment (Filbird) [inline launch.go]: Sounds good! I'm thinking we still want to ensure merge takes a branch argument, so something like the following:  
```
func (r repo) MergePR(prNumber string) error {
    branch := "pr"
	if err := fetchPR(prNumber, branch, r.Path); err != nil {
		return fmt.Errorf("fetching pr: %v", err)
	}

	if err := merge(branch, r.Path); err != nil {
		return fmt.Errorf("merging pr: %v", err)
	}
	return nil
}

```

Comment (jer) [inline launch.go]: Yep, perfect!

File: git/git_test.go [hunk 2]
```diff
@@ -97,417 +58,125 @@ func TestHelperProcess(*testing.T) {
...
-		want := []string{
-			"merge", "--no-edit", testBranch,
-		}
-		if len(args) != len(want) {
-			t.Errorf("Incorrect args sent to git: %q, want %q", args, want)
-		}
-		for i, arg := range args {
-			if arg != want[i] {
-				t.Errorf("args[%d] = %q, want %q", i, arg, want[i])
+			switch args[0] {
```
Comment (jer) [inline git/git_test.go]: Maybe add a default case of t.Errorf here?

Comment (Filbird) [inline git/git_test.go]: Yep, makes sense to do so.

File: git/git.go [hunk 3]
```diff
@@ -9,111 +9,113 @@ import (
...
-// FetchPR fetches a pull request into a specified branch
-func FetchPR(prNumber string, branch string) error {
-	return command("fetch", "origin", "pull/"+prNumber+"/head:"+branch)
-}
-
-// Merge merges changes on the specified branch
-func Merge(branch string) error {
-	return command("merge", "--no-edit", branch)
+	repo := repo{
+		ScmURL: parts[0],
```
Comment (nkatzman) [inline git/git.go]: should we add const values for the 0 and 1 magic numbers?

Something like "SCM_URL_INDEX" and "BRANCH_INDEX"

Comment (Filbird) [inline git/git.go]: good catch! Will make the change

**Comment by jer** [comment]
This is awesome 👍

**Pull request merged**